added chart support for word and powerpoint in addition to excel

dineshkaushal opened the pull request:
This fixes partially #1541. 
This is a common implementation for office charts that works for Excel, word, and powerpoint.
Earlier implementation for excel chart #1987 worked on event based model, but this one provides keyboard navigation which is much more structured. 
excelChart.py is redundent, but I have still not removed it just in case users want event based implementation back.

dineshkaushal committed:
added chart support for word and powerpoint in addition to excel

michaelDCurran commented on source/NVDAObjects/window/winword.py:
```diff
@@ -382,6 +384,20 @@ def label(self):
+		return _(u"{text}").format(text=text)
+
+	def moveTo(self):
+		chartNVDAObj = WordChart(windowHandle=self.document.rootNVDAObject.windowHandle, wordApplicationObject=self.rangeObj.Document.Application, wordShapeObject=self.collectionItem)
```
I think this needs to be changed to OfficeChart

michaelDCurran commented:
It looks like you have not committed _msOfficeChart.py

dineshkaushal committed:
_msOfficeChart.py added

dineshkaushal committed:
changed WordChart to _msOfficeChart.OfficeChart and fixed a problem regarding error in word while processing the formula

dineshkaushal commented:
Oops, I missed that file. I have also made another correction. This is regarding using formula to get text directly from the cell for xValues when formula property has series formula

michaelDCurran commented on source/NVDAObjects/window/_msOfficeChart.py:
```diff
@@ -0,0 +1,972 @@
+# -*- coding: UTF-8 -*-
+#NVDAObjects/window/_msOfficeChartConstants.py
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2014-2015 James Teh, Michael Curren, NV Access Limited
```
Fix the copyright statement.  Something like Copyright (C) 2014-2017 NV access Limited, Dinesh Kaushal

michaelDCurran commented on source/NVDAObjects/window/_msOfficeChart.py:
```diff
@@ -0,0 +1,972 @@
+
+# Dictionary for the Description of chart types.
+chartTypeDict = {
+    # Translators: A type of chart in Microsoft Excel. 
```
Change this and all other comments to mention Microsoft Office, not Microsoft Excel specifically.

michaelDCurran commented on source/NVDAObjects/window/_msOfficeChart.py:
```diff
@@ -0,0 +1,972 @@
+			text +=_("No Series defined.")
+		return text
+
+	def getChartSegment(self):
```
Is this different to OfficeChartElementBase.getChartSegment? if not then this can be removed.

michaelDCurran commented on source/NVDAObjects/window/_msOfficeChart.py:
```diff
@@ -0,0 +1,972 @@
+				"kb:escape": "disablePassThrough",
+	}
+
+class OfficeChartElement(OfficeChartElementList):
```
I find the name of this particular class confusing. I'm assuming that this element represents all the stuff that is not a series? perhaps call it OfficeChartElementMeta or something? saying OfficeChartElement makes it look too generic.

dineshkaushal committed:
changed OfficeChartElement to OfficeChartElementCollection, removed getChartSegment and changed comments from excel to office

michaelDCurran commented:
@dineshkaushal: a reminder to change the copyright line in _msOfficeChart.py to remove Michael Curran and James Teh (NV Access Limited) is enough. Also, can you merge master and fix conflicts again please. thanks :) Otherwise this is looking good.
Finally, it would be great if you could provide some sample office files with charts in them if you have not already. e.g. a Powerpoint slide deck, Excel workbook, and Word doc, all with charts. This will help me do some final testing.

dineshkaushal committed:
Updated to include escape for chart in word and excel

michaelDCurran committed:
Merge branch 'master' into in_t1541_review

michaelDCurran committed:
Add translator comments

michaelDCurran committed:
Excel: make the parent of any chart the active worksheet, so that switching sheets is still correctly supported.
also ensure that charts are not classed as being in Excel's treeInterceptor, so that they can be dismissed properly with escape.

michaelDCurran committed:
Powerpoint appModule: after focusing ona chart in slide view, you now must press enter on the chart to interact with it. This ensures you can still move the chart around the slide with the arrows like any other shape when not interacting.

michaelDCurran committed:
Remove excelChart.py.  No need to keep it here, it is in the git history.

michaelDCurran committed:
Fix copyright statement.

michaelDCurran committed:
Add translator comment

Adriani90 commented:
Dear all,

before creating a new issue, I wanted to ask, is it possible to report if a diagram is concav or convex by signal tones? So when I navigate from value a to b and the diagram is concav, the sound should go higher and NVDA should just read the actual value. We could take the signal tones from reporting progress bars for example.

dineshkaushal commented:
Thanks Mick,

dineshkaushal commented:
In the current implementation we are not reporting the information with tones, but that could be a good thing to try!
